RELEASING.rst

```diff
@@ -0,0 +1,17 @@
+Here are the steps on how to make a new release.
+
+1. Create a ``release-VERSION`` branch from ``upstream/master``.
+2. Install ``bumpversion`` and execute:
+
+   ::
+
+    bumpversion minor --new-version 1.0.11
+
+   Changing ``minor`` and ``--new-version`` above accordingly.
+
+3. Push a branch with the changes.
+4. Once all builds pass and at least another maintainer approves, push a tag to ``upstream`` in the format ``v1.0.11`.
+   This will deploy to PyPI.
+5. Merge the PR (do not squash, to keep the tag).
+6. Create a new release on GitHub, posting the contents of the current CHANGELOG.
```

Comment:
I see change notes are published using GitHub Releases, however I could not find how the changes are recorded/maintained between releases. Here I propose to use a new `CHANGELOG.md` file which is tracked in the repository.

Or previously it was used Draft releases to track changes?

Reply:
@nicoddemus actually I don't recall setting that up, someone else might have. Whatever you find easier is fine by me, like whatever you are using for other projects.

Reply:
> Whatever you find easier is fine by me, like whatever you are using for other projects.

Now that we have one GitHub Release, GitHub will insist to show that as the latest release in the side bar, even if we push later tags:

![image](https://user-images.githubusercontent.com/1085180/231139148-32e703d7-6210-49e8-9953-2ea078313f7b.png)

So I suggest we keep using it, it is simple enough.

> As long as we only have one place for the info, so there might be some info from previous releases to migrate?

I did not find any unfortunately.